workloads/wmt/wmt_jax/submission.py

```diff
@@ -0,0 +1,113 @@
+"""Training algorithm track submission functions for WMT."""
+import functools
+from typing import Iterator, List, Tuple
+
+from . import train
+from . import workload as w
+from flax import jax_utils
+from flax import optim
+from flax.training import common_utils
+import jax
+import numpy as np
+import spec
+
+
+def get_batch_size(workload_name):
+  batch_sizes = {"wmt_jax": w.CONFIG.per_device_batch_size}
+  return batch_sizes[workload_name]
+
+
+def init_optimizer_state(workload: spec.Workload,
+                         model_params: spec.ParameterContainer,
+                         model_state: spec.ModelAuxiliaryState,
+                         hyperparameters: spec.Hyperparamters,
+                         rng: spec.RandomState) -> spec.OptimizerState:
+  del model_state
+  del rng
+
+  optimizer_def = optim.Adam(
+      learning_rate=hyperparameters.learning_rate,
+      beta1=1.0 - hyperparameters.one_minus_beta_1,
+      beta2=0.98,
+      eps=hyperparameters.epsilon)
+  optimizer = optimizer_def.create(model_params)
+
+  # Replicate optimizer.
+  optimizer = jax_utils.replicate(optimizer)
+
+  learning_rate_fn = train.create_learning_rate_scheduler(
+      base_learning_rate=hyperparameters.learning_rate,
+      warmup_steps=workload.config.warmup_steps)
+
+  # compile multidevice versions of train/eval/predict step and cache init fn.
+  p_train_step = jax.pmap(
```

Comment:
It seems that then we need to explicitly deal with replica of optimizers? The existing method doesn't require replica/unreplica optimizers manually.

Reply:
It seems most natural to create p_train_step in initialization rather than update_param so that we don't assign learning_rate_scheduler and config to train_step every time.